horace_core/utilities/split_vector_fixed_sum.m

```diff
@@ -0,0 +1,113 @@
+function [chunks, idxs] = split_vector_fixed_sum(numeric_vector, chunk_sum, cumulative_sum)
+%SPLIT_VECTOR_FIXED_SUM Split the given vector into sub-vectors such that each
+% sub-vector's sum is exactly 'chunk_sum'. If chunk sum does not divide the sum
+% of the input vector, the final chunk's sum will be the remainder.
+%
+% Values in the input vector may be split between chunks.
+%
+% Input:
+% ------
+% numeric_vector   A vector of numeric, non-negative, values.
+% chunk_sum        A positive value specifying the sum for each sub-vector.
+% cumulative_sum   The cumulative sum of 'numeric_vector'. (optional)
```

Comment:
This is an output in the other split_vector function. It may be simpler to calculate it locally in the same way rather than rely on an optional input which may not have been correctly set up

Reply:
The reason for having it as an input argument was because in functions like [`binary_op_double_`](https://github.com/pace-neutrons/Horace/blob/4967048545f7bc12862c25c253f1d7be0ab6442a/horace_core/sqw/PixelData/%40PixelData/private/binary_op_double_.m#L92) we are already calculating the (cumulative) sum for argument validation. So it's more efficient to pass that in, rather than calculate it again.